src/main/java/org/apache/commons/compress/harmony/pack200/BHSDCodec.java

```diff
@@ -308,11 +308,8 @@ public boolean encodes(final long value) {
             }
         } else if (z < 0) {
             // Need to use integer overflow here to represent negatives.
-            if (cardinality < 4294967296L) {
-                z += cardinality;
-            } else {
-                z += 4294967296L; // this value is equal to (1 << 32).
-            }
+            // this value is equal to (1 << 32).
```

Comment:
Wrong comment, I think. `1 << 32` is `4294967296L`.

![image](https://user-images.githubusercontent.com/304786/153921096-1dfa799e-b6ce-4bad-aa39-54d7546a6dc8.png)

After this change it could be interpreted as `z` being `1 << 32`. Maybe change the text to `4294967296 is 1 << 32` (which helps devs with the magic-numbers issue)

Reply:
I think you'd want to use jshell here, not python...

Reply:
True! I have to get used to jshell. Looks like the integer overflow resets the value to `1`. But using longs works the same way as in Python.

![image](https://user-images.githubusercontent.com/304786/153925028-9d9c2d94-209c-4e6e-a136-e3f5776d68cf.png)

Reply:
So... now what?

Reply:
I think the comment needs to be updated to say that `4294967296L` is the `1 << 32`, and not `z += Math.min(cardinality, 4294967296L);`.